Funcky/Extensions/EnumerableExtensions/AverageOrNone.cs

```diff
@@ -0,0 +1,320 @@
+        /// If the sequence only consists of none or is empty it returns None.
+        /// </summary>
+        /// <typeparam name="TSource">The type of the elements of source.</typeparam>
+        /// <param name="source">A sequence of values that are used to calculate an average.</param>
+        /// <param name="selector">A transform function to apply to each element.</param>
+        /// <returns>The average of the sequence of values or None.</returns>
+        [Pure]
+        public static Option<double> AverageOrNone<TSource>(this IEnumerable<TSource> source, Func<TSource, Option<int>> selector)
+            => source.Aggregate(AverageCalculatorDouble.Empty, (calculator, element) => calculator.Add(selector(element))).Average;
+
+        /// <summary>
+        /// Computes the average of a sequence of Int64 values that are obtained by invoking a transform function on each element of the input sequence.
+        /// If the sequence is empty it returns None.
+        /// </summary>
+        /// <typeparam name="TSource">The type of the elements of source.</typeparam>
+        /// <param name="source">A sequence of values that are used to calculate an average.</param>
+        /// <param name="selector">A transform function to apply to each element.</param>
+        /// <returns>The average of the sequence of values or None.</returns>
+        [Pure]
+        public static Option<double> AverageOrNone<TSource>(this IEnumerable<TSource> source, Func<TSource, long> selector)
+            => source.Aggregate(AverageCalculatorDouble.Empty, (calculator, element) => calculator.Add(selector(element))).Average;
+
+        /// <summary>
+        /// Computes the average of a sequence of optional Int64 values that are obtained by invoking a transform function on each element of the input sequence.
+        /// If the sequence only consists of none or is empty it returns None.
+        /// </summary>
+        /// <typeparam name="TSource">The type of the elements of source.</typeparam>
+        /// <param name="source">A sequence of values that are used to calculate an average.</param>
+        /// <param name="selector">A transform function to apply to each element.</param>
+        /// <returns>The average of the sequence of values or None.</returns>
+        [Pure]
+        public static Option<double> AverageOrNone<TSource>(this IEnumerable<TSource> source, Func<TSource, Option<long>> selector)
+            => source.Aggregate(AverageCalculatorDouble.Empty, (calculator, element) => calculator.Add(selector(element))).Average;
+
+        /// <summary>
+        /// Computes the average of a sequence of Single values that are obtained by invoking a transform function on each element of the input sequence.
+        /// If the sequence is empty it returns None.
+        /// The rules for floating point arithmetic apply see remarks for detail.
```

Comment:
Reference to remarks but remarks are missing (applies to the next three overloads too)

Reply:
Fixed in 264e0516 including the Todos